tmt/steps/prepare/feature/epel-disable.yaml

```diff
@@ -10,36 +12,48 @@
 
         # Disable repos of 'epel' if package 'epel-release' is installed
         - name: Detect package 'epel-release' is installed
-          command: rpm -q epel-release
+          ansible.builtin.command: rpm -q epel-release # noqa: command-instead-of-module
           register: result
-          ignore_errors: True
+          ignore_errors: true
+          changed_when: false
 
         - name: Disable repos of 'epel'
-          shell: yum-config-manager --disable epel epel-debuginfo epel-source
           when: result.rc == 0
+          ansible.builtin.command: yum-config-manager --disable epel epel-debuginfo epel-source
+          register: output
+          changed_when: output.rc != 0
 
-    # RHEL N or CentOS Stream N where N >= 8
-    - when: ansible_distribution_major_version|int >= 8
+    # RHEL 8+ or CentOS Stream 8+
+    - name: Disable repos of 'epel' on RHEL 8+ or CentOS Stream 8+
+      when: ansible_distribution_major_version | int >= 8
       block:
         - name: Install 'dnf config-manager'
-          shell: dnf -y install 'dnf-command(config-manager)'
+          ansible.builtin.command: dnf -y install 'dnf-command(config-manager)'
+          register: output
+          changed_when: output.rc != 0
 
         # Disable repos of 'epel' if package 'epel-release' is installed
         - name: Detect package 'epel-release' is installed
-          command: rpm -q epel-release
+          ansible.builtin.command: rpm -q epel-release # noqa: command-instead-of-module
           register: result
-          ignore_errors: True
+          ignore_errors: true
+          changed_when: false
 
         - name: Disable repos of 'epel' if package 'epel-release' is installed
-          shell: dnf config-manager --disable epel epel-debuginfo epel-source
           when: result.rc == 0
+          ansible.builtin.command: dnf config-manager --disable epel epel-debuginfo epel-source
+          register: output
+          changed_when: output.rc != 0
```

Comment:
I don't think we should use `config-manager` to enable/disable the repositories. There is a dedicated ansible module [yum_repository](https://docs.ansible.com/ansible/latest/collections/ansible/builtin/yum_repository_module.html) to take care of this in a much more concise, faster and comfortable way.

Reply:
I'm afraid even the Ansible developers say this won't be supported, see https://github.com/ansible/ansible-modules-extras/issues/2384. TL;DR, `yum_repository` is mostly about creating repositories, if you want to enable or disable some existing ones, use `yum-config-manager` or something similar. Or a module working with INI files.

`yum_repository` requires `state`, `state` is either `present` or `absent`, and `present` requires base URL, metalink or another repository location. It's easy to remove those repositories, but not possible to keep them installed & disabled.

Reply:
Wouldn't https://docs.ansible.com/ansible/latest/collections/ansible/builtin/yum_module.html work here?

Reply:
> Wouldn't https://docs.ansible.com/ansible/latest/collections/ansible/builtin/yum_module.html work here?

Hi @martinhoyer, to enable `epel` repos, looks we can leverage `ansible.builtin.yum` since parameter `enablerepo` can be used and we have to install package `epel-release`.  And to disable `epel` repos, there are two cases:
* a) package `epel-release` is not installed by user, we should do nothing;
* b) package `epel-release` is installed by user, we should disable `epel` repos (via `config-manager`).

@happz, do you prefer to totally leverage `ansible.builtin.yum` to enable `epel` repos (i.e. without `config-manager`)?

Reply:
> Wouldn't https://docs.ansible.com/ansible/latest/collections/ansible/builtin/yum_module.html work here?

Yes, it definitely looks like a much better version of the module. I keep forgetting they have been revamped into `collections`... It seems better to use the module instead of direct calls to config managers.

Reply:
Hmmm, after reading it once more, I got it all wrong. Would `ansible.builtin.yum` really help? Can it enable or disable a repository without taking any other actions? It deals with packages, not repositories, and `enablerepo` key is expected to work in the context of package install/update operation, "Repoid of repositories to enable for the install/update operation. These repos will not persist beyond the transaction.". So, I don't see how we could tell this particular plugin to disable a given repo from now till the end of the guest.

And https://docs.ansible.com/ansible/latest/collections/ansible/builtin/yum_repository_module.html#ansible-collections-ansible-builtin-yum-repository-module also does not seem to be capable of enable/disable *without additional info* - there's only `state: present` or `state: absent`, and `state: present` we might use for `enabled: false` requires at least one of `baseurl`, `mirrorlist` or `metalink`, and `description` as well, which to me sounds like we'd have to read these first from the repository, and then pass it to `yum_repository` just to get it disabled... A config manager commands seems like a better way to me.

Reply:
My bad, sorry for the noise. I saw enable/disable repo and didn't realize it's only for install/update operations. 
Indeed yum_repository is quite a hassle and I also didn't find a way to utilize it in a practical way for this use-case and agree config-manager command being a better way.